### PR TITLE
needless_doc_main: only check rust code

### LINKED ISSUE #5280: needless_doctest_main false positive with ```txt
This piece of documentation triggers the lint
```rust
/// ```txt
/// 1 |   fn main() {
///   |          __ ^
///   |  _________|_|
///   | |         |
///   | |         this is a pair of parenthesis
/// 2 | |     println!("Hello World!")
///   | |             |^^^^^^^^^^^^^^| this is a pair of parenthesis
///   | |             |_____________||
///   | |                           |
///   | |                           this is a string. Hard to see where it starts, uh?
/// 3 | | }
///   | |_^ this is a pair of braces
/// ```
```

```
warning: needless `fn main` in doctest
   --> src/fmt.rs:207:4
    |
207 | /// 1 |   fn main() {
    |    ^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_doctest_main
```

[`playground`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=%23!%5Bdeny(clippy%3A%3Aneedless_doctest_main)%5D%0A%0A%2F%2F%2F%20%60%60%60txt%0A%2F%2F%2F%201%20%7C%20%20%20fn%20main()%20%7B%0A%2F%2F%2F%20%20%20%7C%20%20%20%20%20%20%20%20%20%20__%20%5E%0A%2F%2F%2F%20%20%20%7C%20%20_________%7C_%7C%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20%7C%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20this%20is%20a%20pair%20of%20parenthesis%0A%2F%2F%2F%202%20%7C%20%7C%20%20%20%20%20println!(%22Hello%20World!%22)%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20%20%20%20%20%7C%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%7C%20this%20is%20a%20pair%20of%20parenthesis%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20%20%20%20%20%7C_____________%7C%7C%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7C%0A%2F%2F%2F%20%20%20%7C%20%7C%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20this%20is%20a%20string.%20Hard%20to%20see%20where%20it%20starts%2C%20uh%3F%0A%2F%2F%2F%203%20%7C%20%7C%20%7D%0A%2F%2F%2F%20%20%20%7C%20%7C_%5E%20this%20is%20a%20pair%20of%20braces%0A%2F%2F%2F%20%60%60%60%0Apub%20fn%20example()%20%7B%7D%0A%0Afn%20main()%20%7B%7D%0A)

### PR DESCRIPTION
**Pull request by llogiq**
This fixes #5280 by checking the language attribute on code blocks.

---

changelog: none

### CODE REVIEW

**Comment by llogiq** [comment]
Also @Manishearth can we remove the Appveyor build from the repo?

File: tests/ui/needless_doc_main.rs [hunk 1]
```diff
@@ -34,9 +50,25 @@ fn bad_doctest() {}
 ///     assert_eq(1u8, test::black_box(1));
 /// }
 /// ```
+///
+/// We should not lint non-rust examples:
```
Comment (flip1995) [inline tests/ui/needless_doc_main.rs]:
```suggestion
/// We should not lint ignored examples:
```

Comment (flip1995) [reply tests/ui/needless_doc_main.rs]: r=me after this

**Comment by Manishearth** [comment]
Done

**Comment by Manishearth** [comment]
I didn't delete the project so we can keep the history, i just made it not trigger builds for anything anymore

**Pull request merged**